src/main/java/org/springframework/data/couchbase/core/convert/CryptoConverter.java

```diff
@@ -81,22 +82,12 @@ private Object coerceToValueRead(byte[] decrypted, CouchbaseConversionContext co
 		CouchbasePersistentProperty property = context.getProperty();
 
 		CustomConversions cnvs = context.getConverter().getConversions();
-		ConversionService svc = context.getConverter().getConversionService();
 		Class<?> type = property.getType();
 
 		String decryptedString = new String(decrypted);
 		if ("null".equals(decryptedString)) {
 			return null;
 		}
-		// TODO - as-is, this never gets ran through ObjectMapper() -
-		// TODO - i.e. @JsonValue etc will not be processed by ObjectMapper
-
-		/* this what we would do if we could use a JsonParser with a beanPropertyTypeRef
-		    final JsonParser plaintextParser = p.getCodec().getFactory().createParser(plaintext);
-		plaintextParser.setCodec(p.getCodec());
-		
-		return plaintextParser.readValueAs(beanPropertyTypeRef);
-		 */
 
```

Comment:
The deleted comment is a bit misleading.  The conversions do take place in context.getConverter.read() and getPotentiallyConvertedSimpleRead() below.